CMakeLists.txt

```diff
@@ -41,11 +41,20 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 include(CMakeDependentOption)
 
-message(STATUS "pybind11 v${pybind11_VERSION} ${pybind11_VERSION_TYPE}")
+if(NOT pybind11_QUIETLY)
```

Comment:
It much match the project name. I did try changing the project name to `PYBIND11`, and I've forgotten now why I didn't keep it. It might have been that some of the generated files have `${PROJECT}` or some form of that in the name, but that could easily be changed. I'll revisit this.

Reply:
Ah, part of it / most if it is due to `find_package(pybind11` - the variables it sets match the case of the package you find - `pybind11_FOUND`, for example. From that, you expect (and this is the way it used to work, too) `pybind11_VERSION`, `pybind11_INCLUDE_DIRS`, etc.

Reply:
I am all in favour of consistent casing. 
CMake uses `<PackageName>_FIND_QUIETLY` for finder-modules. I am failing to see a connection to `pybind11_QUIETLY`. What am I missing?

Reply:
:confounded:

You are right. I meant `pybind11_FIND_QUIETLY`. (I'm allowing it to be set even in add_submodule mode, but it should be consistent)

Reply:
Config mode sets these variables:

```
pybind11_FOUND - true if pybind11 and all required components found on the system
  pybind11_VERSION - pybind11 version in format Major.Minor.Release
  pybind11_INCLUDE_DIRS - Directories where pybind11 and python headers are located.
  pybind11_INCLUDE_DIR - Directory where pybind11 headers are located.
  pybind11_DEFINITIONS - Definitions necessary to use pybind11, namely USING_pybind11.
  pybind11_LIBRARIES - compile flags and python libraries (as needed) to link against.
  pybind11_LIBRARY - **empty.**
```
Shouldn't we match them in add_subdirectory mode? Currently add_subdirectory mode sets uppercase version of `PYBIND11_INCLUDE_DIR`,  and several of the others seem to be missing. `PYBIND11_*` could be configuration (input) options, and `pybind11_*` could be output.

Reply:
(Doesn't have to be done in this PR, though)

Reply:
I've added FOUND and INCLUDE_DIR. `pybind11_DEFINITIONS` doesn't really seem useful (the `USING_pybind11` is used anywhere, and isn't included anywhere), and I'd rather like to drop it. The others are generally not things I'd like to push for (LIBRARIES and INCLUDE_DIRS), since targets are _much_ better (and you can extract this info if needed).